ImageTaggingApp.Console.App/Services/ImageTaggingService.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+using ImageTaggingApp.Console.App.APIs;
+using ImageTaggingApp.Console.App.Entities;
+
+namespace ImageTaggingApp.Console.App.Services {
+    public interface IImageTaggingService {
+        IList<Image> Tag(IReadOnlyCollection<string> imagesPaths);
+    }
+
+    public class ImageTaggingService : IImageTaggingService {
+        private readonly IImageTaggingApi _imageTaggingApi;
+
+        public ImageTaggingService(IImageTaggingApi imageTaggingApi) {
+            _imageTaggingApi = imageTaggingApi;
+        }
+
+        public IList<Image> Tag(IReadOnlyCollection<string> imagesPaths) {
+            if (imagesPaths.Count == 0) {
+                throw new ArgumentException("Image paths count is equal to zero.", nameof(imagesPaths));
+            }
+            var taggedImages = new List<Image>();
+            foreach (var imagePath in imagesPaths) {
```

Comment:
1.
- Agree, we need to track progress. Found https://gist.github.com/DanielSWolf/0ab6a96899cc5377bf54 solution, will try to implement.
-  Currently added ability to cancel command using ctrl+c combination. I don't know what other approaches might be for console app
2. I think we should track API calls limit in the class which implements IImageTaggingApi interface. Api limits for minute/day/month differs on API provider. 
For user to choose how to handle API limits, we need to introduce some sort of configuration in config file or a new command parameter where user chooses from possible choices.